### PR TITLE
add: tests for firestore adapter

### PR DESCRIPTION
**Pull request by AlexMeuer**
Ported the Prisma tests.

Relates to #18

### CODE REVIEW

File: packages/firebase/tests/index.test.ts [hunk 1]
```diff
@@ -1 +1,153 @@
...
+            {
+                ...session,
+                expires: expiresRequested,
+            },
+            true
+        )) as ISession;
+
+        const difference = Math.abs(session.expires.getTime() - expiresExpected);
+        expect(difference).toBeLessThan(5);
+    });
```
Comment (AlexMeuer) [inline packages/firebase/tests/index.test.ts:95]: Unlike Prisma, the Firestore adapter does not return the session with the given expiry time. This is the test I'm least sure about.

**Comment by ndom91** [comment]
@AlexMeuer thanks a lot for getting this done! One problem though, did you have `ts-mock-firebase` installed locally? I wasn't included in the package.json, but is required by some of your tests.

I tried to install myself, but unfortunately it won't compile on my linux install, its trying to compile from source this `grpc` library from python I think, so theres all sorts of node-gyp errors. There hasn't been a release in 2 years for `ts-mock-firebase` unfortunately so I assume some versions its requiring are too old / incompatible with current python / node-gyp / whatever

**Comment by AlexMeuer** [comment]
Hmm, I must have forgotten to save the dependency. I'll have a look for an alternative when I've time.

**Comment by AlexMeuer** [comment]
Spent some time trying to replace `ts-mock-firebase` with `firebase-mock` but it lacks support for serverTimestamp as well as screwing with async/await. Might just have to manually mock the used methods, but it doesn't look like I'll have more time this week.

**Comment by ndom91** [comment]
Hey so first of all, thanks for giving it a shot and reporting back on any and all stumbling blocks! 

I'll take a shot at specifically mocking the used methods then. Dom have you pushed your latest state to this branch?

**Comment by balazsorban44** [comment]
Thank you for this, but I think https://github.com/nextauthjs/adapters/pull/90 will address most of this, and also some of the stuff here has been made redundant by improving our core package. It takes care of error handling instead of the adapters.

**Pull request closed without merging**